mlrun/utils/clones.py

```diff
@@ -126,8 +126,24 @@ def get_secret(key):
 
     branch = None
     tag = None
+    commit_id = None
     if url_obj.fragment:
         refs = url_obj.fragment
+        # the intended format for submitting commit id is
+        # git:/github.com/example/myproject.git#refs/heads/mybranch#desired-commit-id> or
+        # git:/github.com/example/myproject.git#mybranch#desired-commit-id>.
+        # we updated this format to comply to kaniko conventions.
+        # For further details, see https://github.com/GoogleContainerTools/kaniko#kaniko-build-contexts.
+        if "#" in refs:
+            try:
+                refs, commit_id = refs.split("#")
+            except ValueError:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"Unable to resolve commit id from {url},"
+                    f" expecting url format as following: "
+                    f"git://github.com/example/myproject.git#refs/heads/mybranch#<desired-commit-id> or"
+                    f"git:/github.com/example/myproject.git#mybranch#desired-commit-id>"
```

Comment:
Using a url in this format (containing 2 `#`s) will have 3 values to unpack when you expect 2 and will raise an exception. Either there's a mistake in the url, or in the handling of the split